### DISCUSSION TITLE
Does fetching data from DB with the js library perform a token refresh if necessary?

### DISCUSSION THREAD

**Discussion by astonfuture** [supabase/supabase#19579] [Questions] [answered]
If I was on the server, using a createServerClient supabase client, would using the following command refresh the auth token if it was expired and set a Set-Cookie header on the response object?

`supabase.from('my_table').select()`

Or do you explicitly need to call supabase.auth.getSession() to force a refresh?

**Comment by GaryAustin1** [comment, marked as answer]
No.   Normally the browser client keeps the token and cookies refreshed based on a timer if you are using auth-helpers/ssr (based on createServerClient I assume you are).  If you are not using that library then you would need to make sure the session is refreshed and getSession would do that if it is expired.
